cmd/rpc/client.go

```diff
@@ -60,29 +61,37 @@ func (client *Client) Call(serviceMethod string, args, reply interface{}) error
 
 	var buf bytes.Buffer
 	if err = gob.NewEncoder(&buf).Encode(callRequest); err != nil {
-		return err
+		return nil, err
 	}
 
-	response, err := client.httpClient.Post(client.serviceURL.String(), "", &buf)
+	var body io.Reader = &buf
+	if reader != nil {
+		body = io.MultiReader(body, reader)
```

Comment:
Unable to understand the purpose of this, why are we concatenating the gobencoded data with additional data?

Reply:
This way any request stream with request args will be sent to the server i.e. there is no need to have any special method to send request reader.

Reply:
Ah, I see what you are trying to do, that's clever.

Reply:
But note that the PR is still incomplete, i.e it has just added support for streams at rpc level, we still need to take advantage of it for the StorageAPI calls like ReadFile